R/rmarkdown/knit.R

```diff
@@ -11,7 +11,9 @@ knit_command <- Sys.getenv("VSCR_KNIT_COMMAND")
 
 # set the knitr chunk eval directory
 # mainly affects source calls
-knitr::opts_knit[["set"]](root.dir = knit_dir)
+if (!is.null(eval(parse(text = knit_dir)))) {
```

Comment:
When `knit_dir` is a path like `/home/user/documents`, then `eval(parse(text = knit_dir))` will result in an error

```
Error in parse(text = knit_dir) : <text>:1:1: unexpected '/'
1: /
    ^
Calls: eval -> parse

Backtrace:

1: parse(text = knit_dir)
2: eval(parse(text = knit_dir))

Exiting on error
```

Reply:
I think `knitWorkingDirText` should fallback to empty string `""` and checking `nzchar(knit_dir)` should be enough.